### PR TITLE
Converted to Swift4 and did some cleaups

### PR DESCRIPTION
**Pull request by crsantos**
## Changes

* Converts project to swift 4.1 on both targets
* Removed some useless force unwraps and added some `guard` instead
* Moved some private methods into a separate `private extension`

### CODE REVIEW

File: UnicornLeap/Command.swift [hunk 1]
```diff
@@ -11,7 +11,7 @@ class Command {
 
   fileprivate func flagIndex(_ flags: [String]) -> Int? {
     let indexes = flags.map { arguments.index(of: $0) }
-    return indexes.flatMap({$0}).first
+    return indexes.compactMap{ $0 }.first
```
Comment (jonallured) [inline UnicornLeap/Command.swift]: 😎

File: UnicornLeap/Leap.swift [hunk 2]
```diff
@@ -13,9 +13,14 @@ class Leap {
...
-    NSApplication.shared()
+    _ = NSApplication.shared
 
-    let floatingWindow = FloatingWindow(rect: NSScreen.main()!.frame)
+    guard let mainScreen = NSScreen.main else {
+
+        fatalError("Error unwrapping NSScreen.main")
+    }
+
+    let floatingWindow = FloatingWindow(rect: mainScreen.frame)
```
Comment (jonallured) [inline UnicornLeap/Leap.swift]: Good call, this is better. 👍

File: UnicornLeap/main.swift [hunk 3]
```diff
@@ -41,8 +41,8 @@ func printImageError(_ filename: String) {
 }
 
 func printVerboseOutput() {
-  print("Seconds: \(command.seconds)")
-  print("Number: \(command.number)")
+  print("Seconds: \(String(describing: command.seconds))")
+  print("Number: \(String(describing: command.number))")
```
Comment (jonallured) [inline UnicornLeap/main.swift]: Doesn't the string interp essentially do this for us? I guess I assumed that `command.number` was converted to a string implicitly here, but maybe I'm missing something!

**Review by jonallured — APPROVED** [review]
Nice one!! I'm curious why you liked putting `configurePath` and `configureLayer` in a private extension over having them be private funcs but that's just for my own info, this is good to merge. ❤️

File: UnicornLeap/main.swift (hunk 3 above)
Comment (crsantos) [reply UnicornLeap/main.swift]: As it is an `Optional` it requires that to silence the warning 😄

**Comment by crsantos** [comment]
> Nice one!! I'm curious why you liked putting configurePath and configureLayer in a private extension over having them be private funcs

Well, it's just a matter of organization. 
- `private` methods into private extensions
- Each method of a `protocol` conformance into extensions 😄

**Comment by crsantos** [comment]
Thanks for reviewing

**Comment by jonallured** [comment]
Looks like this is stale so closing but thanks for the PR!! ❤️

**Comment by crsantos** [comment]
well, it was not stale.

It was ready to merge at the time, I guess.
But I couldn't merge without your approval.

**Comment by jonallured** [comment]
@crsantos sorry I guess what I was trying to say is that with this one: https://github.com/jgdavey/unicornleap/commit/7dc3a29593d636374b946adfd834d53ca8053414 the project has been upgraded to Swift 5 so I think that probably means there's not much to do here. Thanks again!

**Comment by crsantos** [comment]
Nice! Happy that it’s upgraded 🤗

**Pull request closed without merging**